packages/frontend-web/src/app/scenes/Tables/components.tsx

```diff
@@ -0,0 +1,268 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { autobind } from 'core-decorators';
+import { Set } from 'immutable';
+import React from 'react';
+import Timer = NodeJS.Timer;
+
+import { IArticleModel, IUserModel } from '../../../models';
+import * as icons from '../../components/Icons';
+import { GREY_COLOR, NICE_CONTROL_BLUE, NICE_MIDDLE_BLUE } from '../../styles';
+import { css } from '../../util';
+import { COMMON_STYLES, ICON_STYLES } from './styles';
+
+export interface IMagicTimestampProps {
+  timestamp: string;
+  inFuture: boolean;
+}
+
+export class MagicTimestamp extends React.Component<IMagicTimestampProps> {
```

Comment:
This seems like some quite intricate logic - maybe add a test? And/or documentation to explain a bit more what this is intended to do.

Reply:
SGTM